contracts/core/Modifier.sol

```diff
@@ -73,7 +74,18 @@ abstract contract Modifier is Module, IAvatar {
     */
 
     modifier moduleOnly() {
-        if (modules[msg.sender] == address(0)) revert NotAuthorized(msg.sender);
+        bool wasSent = modules[msg.sender] != address(0);
+        bool wasSigned = !wasSent && modules[eip712SignedBy()] != address(0);
```

Comment:
I believe you could save gas by bundling this all into one logical statement with no assignment.

i.e. there is no point checking and assigning`wasSigned` if `wasSent` is true.

Reply:
Originally this was written like in the picture:

<img width="545" alt="Screenshot 2023-09-28 at 05 28 02" src="https://github.com/gnosis/zodiac/assets/12870300/5472638e-4068-46c4-ad23-18c32b0b8fe1">

But since it's such a security critical chunk of code, I opted for readability.

Should we go with the gas optimized way?

Reply:
Yeah, I'd save the gas. I don't find it any less readable.

Reply:
Sorted